keras_cv/models/object_detection/retina_net/retina_net_inference_test.py

```diff
@@ -30,46 +30,79 @@ def cleanup_global_session(self):
 
     def test_weight_setting(self):
         x, y = _create_bounding_box_dataset(bounding_box_format="xywh")
-        retina_net, new_retina_net = _create_retina_nets(x, y, epochs=1)
+        pretrained_retina_net, new_retina_net = _create_retina_nets(x, y, epochs=1)
 
-        new_retina_net.set_weights(retina_net.get_weights())
+        new_retina_net.set_weights(pretrained_retina_net.get_weights())
 
         # check if all weights that show up via `get_weights()` are loaded
         for retina_net_weight, post_load_weight in zip(
-            retina_net.get_weights(), new_retina_net.get_weights()
+            pretrained_retina_net.get_weights(), new_retina_net.get_weights()
         ):
             self.assertAllEqual(retina_net_weight, post_load_weight)
 
         for layer_original, layer_new in zip(
             # manually check layers to make sure nothing is missed
-            _get_retina_net_layers(retina_net),
+            _get_retina_net_layers(pretrained_retina_net),
             _get_retina_net_layers(new_retina_net),
         ):
             for weight, weight_new in zip(
                 layer_original.get_weights(), layer_new.get_weights()
             ):
                 self.assertAllEqual(weight, weight_new)
 
+    def test_decoder_doesnt_get_updated(self):
+        x, y = _create_bounding_box_dataset(bounding_box_format="xywh")
+        pretrained_retina_net, new_retina_net = _create_retina_nets(
+            x, y, epochs=1, custom_decoder=True
+        )
+        new_retina_net.set_weights(pretrained_retina_net.get_weights())
+
+        # check if all weights that show up via `get_weights()` are loaded
+        for retina_net_weight, post_load_weight in zip(
+            pretrained_retina_net.get_weights(), new_retina_net.get_weights()
+        ):
+            self.assertAllEqual(retina_net_weight, post_load_weight)
+
+        pretrained_decoder = pretrained_retina_net.prediction_decoder
+        new_decoder = new_retina_net.prediction_decoder
+        self.assertEqual(new_decoder.suppression_layer.iou_threshold, 0.75)
+        self.assertNotEqual(
+            new_decoder.suppression_layer.iou_threshold,
+            pretrained_decoder.suppression_layer.iou_threshold,
+        )
+
+    def test_savedmodel_creation(self):
+        x, y = _create_bounding_box_dataset(bounding_box_format="xywh")
+        pretrained_retina_net, new_retina_net = _create_retina_nets(x, y, epochs=1)
+
+        tmp = tempfile.mkdtemp()
+        pretrained_retina_net.save(f"{tmp}/checkpoint/")
+        load_model = tf.saved_model.load(f"{tmp}/checkpoint/")
```

Comment:
I think you can either use this, or `new_retina_net.load_weights(f"{tmp}/checkpoint/")`.

For the returned result from model.load_weights(), you can assert all the weights have consumed, eg assert_consumed().

https://www.tensorflow.org/api_docs/python/tf/train/Checkpoint#restore